lib/spack/spack/parse.py

```diff
@@ -122,11 +122,11 @@ def accept(self, id):
 
     def next_token_error(self, message):
         """Raise an error about the next token in the stream."""
-        raise ParseError(message, self.text, self.token.end)
+        raise ParseError(message, self.text[0], self.token.end)
```

Comment:
Is `self.text[0]` always guaranteed to be populated (and you won't get index out of bounds error?)

Reply:
it is -- if it's parsing the parsed text has to come from there.